tangelo/linq/target/target_cirq.py

```diff
@@ -59,21 +59,23 @@ def simulate_circuit(self, source_circuit: Circuit, return_statevector=False, in
             numpy.array: The statevector, if available for the target backend
                 and requested by the user (if not, set to None).
         """
-
-        translated_circuit = translate_c(source_circuit, "cirq",
-            output_options={"noise_model": self._noise_model})
-
-        if source_circuit.is_mixed_state or self._noise_model:
-            # Only DensityMatrixSimulator handles noise well, can use Simulator but it is slower
+        n_meas = source_circuit._gate_counts.get("MEASURE", 0)
+        if (source_circuit.is_mixed_state or self._noise_model) and not save_mid_circuit_meas:
+            # Only DensityMatrixSimulator handles noise well, can use Simulator, but it is slower
             cirq_simulator = self.cirq.DensityMatrixSimulator(dtype=np.complex128)
+        elif save_mid_circuit_meas:
+            cirq_simulator = self.cirq.DensityMatrixSimulator(dtype=np.complex128) if self._noise_model else self.cirq.Simulator(dtype=np.complex128)
```

Comment:
I assume I wrote this but I'm not sure these if statements are correct in retrospect. 
But what should happen is.
`DensityMatrixSimulator` should only be used when 
1) `self._noise_model`
2) `self._is_mixed_state` and not `save_mid_circuit_meas`

Otherwise.`Simulator` should be used.

Reply:
Do you mean then:
```
if self._noise_model or (source_circuit.is_mixed_state and not save_mid_circuit_meas): 
    cirq_simulator = self.cirq.DensityMatrixSimulator(dtype=np.complex128)
else:
    cirq_simulator = self.cirq.Simulator(dtype=np.complex128)
```
Actually, looks like that's what it's doing anyway, in a roundabout way.

Reply:
Yes